scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -73,6 +82,158 @@ public void visit(CheckExpressionNode checkExpressionNode) {
     }
 
     @Override
+    public void visit(BinaryExpressionNode binaryExpressionNode) {
+        reportIssuesForTrivialOperations(binaryExpressionNode);
+        isSameOperandRestrictedInOperator(binaryExpressionNode.operator()).ifPresent(rule -> {
+            checkSameUsageInBinaryOperator(binaryExpressionNode.lhsExpr(),
+                    binaryExpressionNode.rhsExpr(), rule, binaryExpressionNode);
+        });
+    }
+
+    @Override
+    public void visit(AssignmentStatementNode assignmentStatementNode) {
+        checkSameUsageInAssignment(assignmentStatementNode.varRef(), assignmentStatementNode.expression(),
+                CoreRule.SELF_ASSIGNMENT, assignmentStatementNode);
+        this.visitSyntaxNode(assignmentStatementNode);
+    }
+
+    @Override
+    public void visit(CompoundAssignmentStatementNode compoundAssignmentStatementNode) {
+        checkSameUsageInAssignment(compoundAssignmentStatementNode.lhsExpression(),
+                compoundAssignmentStatementNode.rhsExpression(), CoreRule.SELF_ASSIGNMENT,
+                compoundAssignmentStatementNode);
+        this.visitSyntaxNode(compoundAssignmentStatementNode);
+    }
+
+    private void checkSameUsageInBinaryOperator(Node lhs, Node rhs, CoreRule rule, Node node) {
+        if (isSameSimpleExpression(lhs, rhs)) {
+            reportIssue(node, rule);
+        }
+    }
+
+    private void checkSameUsageInAssignment(Node lhs, Node rhs, CoreRule rule, Node node) {
+        if (isSameSimpleExpression(lhs, rhs)) {
+            reportIssue(node, rule);
+        }
+    }
+
+    public void reportIssuesForTrivialOperations(BinaryExpressionNode binaryExpressionNode) {
```

Comment:
How do we define trivial?

Reply:
Let's rethink the name. Also doesn't have to be public?

Reply:
Trivial means, by definition there are defined as it is.
ex:- any number value is less than infinity